### DISCUSSION TITLE
Solana Sign Transaction - Signing throught solana Network

### DISCUSSION THREAD

**Discussion by daniellnog** [WalletConnect/walletconnect-monorepo#2324] [Developer Support]
Hey guys, good afternoon.

I'm migrating my wallet app from v1 to v2, and, also, developing solana integration.

My sign message is already working well, but, signing transaction is not working as expected.

First of all, I don't know if it happen just with me, but, your dapp demo (https://react-app.walletconnect.com/) seems is not working for solana sign transaction. I receive:

`failed to get recent blockhash: Error: 403 : {"jsonrpc":"2.0","error":{"code": 403, "message":"Access forbidden, contact your app developer or support@rpcpool.com."}, "id": "e1afc592-91d8-4cfb-824c-7230fcd65f6b" }`.

Then, I went throught your explore section and find a dapp working in v2 and solana. Then, Starting developing this stuff, but, after signing process throught solana network, I receive the signature in buffer format, and then, convert it to bs58 to send it back to wallet connect, but, it's not working....I'm receiving a message `Something went wrong
Signature verification failed`, but, from solana network, nothing wrong is comming, so, I would say probably is something related to Wallet Connect. 

Is needed to send the transaction to solana network with the sendRawTransaction, and then send it back to Wallet Connect? Any tips or suggestions?

Thanks,
Daniel Nogueira

**Comment by finessevanes** [comment]
Hey - let me get this to one of our devs!

**Comment by devceline** [comment]
Hey there! Since you're still early in development, consider using [solib](https://github.com/walletconnect/solib), docs [here](https://solib.dev) I just tested the solib example and everything's working great there for transactions and message signing.

Meanwhile, I'll be investigating the error you posted

**Reply by daniellnog** [reply]
Hi @devceline , questions related to solib, where is the best place to do? :)

**Reply by finessevanes** [reply]
Feel free to drop them in [general-support](https://github.com/orgs/WalletConnect/discussions/categories/general-support).

**Reply by daniellnog** [reply]
thanks @finessevanes . I did it already.

@devceline any news about the error I've posted? 

Thanks!

**Comment by devceline** [comment]
What RPC provider are you using that is returning that `403`? Can we see the entire request as a `cURL` please?

**Reply by daniellnog** [reply]
Yes, there is:

`curl 'https://api.mainnet-beta.solana.com/' \
  -H 'authority: api.mainnet-beta.solana.com' \
  -H 'accept: */*' \
  -H 'accept-language: pt-BR,pt;q=0.9,fr-CA;q=0.8,fr;q=0.7,en-US;q=0.6,en;q=0.5' \
  -H 'cache-control: no-cache' \
  -H 'content-type: application/json' \
  -H 'origin: https://react-app.walletconnect.com' \
  -H 'pragma: no-cache' \
  -H 'referer: https://react-app.walletconnect.com/' \
  -H 'sec-ch-ua: "Google Chrome";v="113", "Chromium";v="113", "Not-A.Brand";v="24"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Windows"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: cross-site' \
  -H 'solana-client: js/0.0.0-development' \
  -H 'user-agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36' \
  --data-raw '{"method":"getRecentBlockhash","jsonrpc":"2.0","params":[],"id":"a3ccea8b-e544-41a0-a095-2bca9be5e28a"}' \
  --compressed`

**Reply by daniellnog** [reply]
any news, @devceline ? :X sorry to bother you :/

**Reply by finessevanes** [reply]
Ping'd @devceline internally. Thank you for your patience!

**Reply by daniellnog** [reply]
Any news guys? I still need some help here :(

**Reply by devceline** [reply]
Discussing with the team on how we can best help - should have an answer soon. Sorry for the delay :pray:

**Reply by daniellnog** [reply]
@devceline @finessevanes some news? 

I'm trying to sign the transaction throught [solana-wallet](https://github.com/WalletConnect/solana-wallet). My problem now is that I'm getting Assertion failed error.  Is something related to signatures array, but, couldn't figure out what can be wrong here. 
<img width="627" alt="Screenshot 2023-06-20 at 01 38 18" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/14362997/0cdb2a7e-609b-43b3-8087-0de3a4b4860e">

The addSignature function receives 2 parameters, where the second one should be the signature. But, in the solana-wallet is using the pubkey. Is that correct?
<img width="753" alt="Screenshot 2023-06-20 at 01 48 25" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/14362997/6d38a7d3-91f9-4db3-ad51-855264233212">

<img width="1015" alt="Screenshot 2023-06-20 at 01 49 39" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/14362997/01aa9fc6-7d4b-47ac-911b-78a6645cbf26">

**Comment by BartonMila** [comment]
Hope to see a solution there too

**Comment by mooooooooony** [comment]
hope you can solve this problem

**Reply by daniellnog** [reply]
Unfortunately, still waiting for response...

**Comment by ganchoradkov** [comment]
hey, the example app was updated couple of days ago with working solana rpc provider https://react-app.walletconnect.com/
please let us know if the issue persists

**Reply by daniellnog** [reply]
It's working. Thanks.

**Reply by daniellnog** [reply]
@ganchoradkov do you also works in the react-app?

I'm asking because I'm having some problems when I have transactions with multiple instructions/signatures that requires more than one signature to get it signed. 

I'm using the solana-wallet package, but, when I pass the array of partialSignatures, it does not work...

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/14362997/d4330f71-c7d1-4fd9-a652-4a65ff92fb89)

Going through the code, deserialiseTransaction, looks like addSignature method is being called passing pubkey in both parameters, but, if I'm not wrong, it should be pubkey, signature instead pubkey, pubkey.

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/14362997/2dd898aa-9de6-4f42-8442-3f75a2d16b67)

**Comment by gavgav228** [comment]
i love solana

**Comment by airose007** [comment]
If you looking for help with building a sniper bot, join here https://t.me/+U_BCU0XDRYU0ZWQy
